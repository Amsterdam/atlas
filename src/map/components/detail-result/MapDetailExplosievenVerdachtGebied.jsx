import React from 'react';
import PropTypes from 'prop-types';

const MapDetailExplosievenVerdachtGebied = ({ panoUrl, verdachtGebied }) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <img
          alt="Panoramabeeld"
          className="map-detail-result__header-pano"
          src={panoUrl}
        />
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">Verdacht gebied</h1>
        <h2 className="map-detail-result__header-subtitle">{verdachtGebied._display}</h2>
      </div>
    </header>
  </section>
);

MapDetailExplosievenVerdachtGebied.defaultProps = {
  panoUrl: ''
};

MapDetailExplosievenVerdachtGebied.propTypes = {
  verdachtGebied: PropTypes.shape({
    label: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string
};

export default MapDetailExplosievenVerdachtGebied;
