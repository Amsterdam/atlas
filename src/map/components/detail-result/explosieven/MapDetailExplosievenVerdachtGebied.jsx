import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';

const MapDetailExplosievenVerdachtGebied = ({ panoUrl, verdachtGebied }) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <img
          alt="Panoramabeeld"
          className="map-detail-result__header-pano"
          src={panoUrl}
        />
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">Verdacht gebied</h1>
        <h2 className="map-detail-result__header-subtitle">{verdachtGebied.label}</h2>
      </div>
    </header>
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Hoofdgroep"
        value={verdachtGebied.type}
      />
      <MapDetailResultItem
        label="Subsoort"
        value={verdachtGebied.subType}
      />
      <MapDetailResultItem
        label="Opmerkingen"
        value={verdachtGebied.remarks}
      />
    </ul>
  </section>
);

MapDetailExplosievenVerdachtGebied.defaultProps = {
  panoUrl: ''
};

MapDetailExplosievenVerdachtGebied.propTypes = {
  verdachtGebied: PropTypes.shape({
    label: PropTypes.string,
    remarks: PropTypes.string,
    subType: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string
};

export default MapDetailExplosievenVerdachtGebied;
