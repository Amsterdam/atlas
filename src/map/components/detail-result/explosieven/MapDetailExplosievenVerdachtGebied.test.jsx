import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenVerdachtGebied from './MapDetailExplosievenVerdachtGebied';

describe('MapDetailExplosievenVerdachtGebied', () => {
  it('should render everything', () => {
    const verdachtGebied = {
      label: 'Verdacht gebied label',
      remarks: 'Verdacht gebied remarks',
      subType: 'Verdacht gebied sub type',
      type: 'Verdacht gebied type'
    };
    const wrapper = shallow(
      <MapDetailExplosievenVerdachtGebied
        panoUrl="panoUrl"
        verdachtGebied={verdachtGebied}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
