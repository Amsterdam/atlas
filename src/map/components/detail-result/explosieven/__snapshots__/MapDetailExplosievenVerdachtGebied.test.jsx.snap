// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenVerdachtGebied should render everything 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verdacht gebied label"
  title="Verdacht gebied"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      hasMultiline={false}
      label="Hoofdgroep"
      link={false}
      value="Verdacht gebied type"
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Subsoort"
      link={false}
      value="Verdacht gebied sub type"
    />
    <MapDetailResultItem
      hasMultiline={true}
      label="Opmerkingen"
      link={false}
      value="Verdacht gebied remarks"
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;
